js_modules/dagster-ui/packages/ui-core/src/app/LayoutProvider.tsx

```diff
@@ -15,41 +34,64 @@ export const LayoutContext = React.createContext<LayoutContextValue>({
   nav: {
     canOpen: true,
     isOpen: false,
+    isSmallScreen: false,
     open: () => {},
     close: () => {},
     setCanOpen: (_canOpen: boolean) => {},
   },
 });
 
+const STORAGE_KEY = 'large-screen-nav-open';
+
 export const LayoutProvider = (props: {children: React.ReactNode}) => {
-  const [navOpen, setNavOpen] = React.useState(false);
+  const [navOpenIfLargeScreen, setNavOpenIfLargeScreen] = useStateWithStorage(
+    STORAGE_KEY,
+    (json: any) => {
+      if (typeof json !== 'boolean') {
+        return false;
+      }
+      return json;
+    },
+  );
+
+  const [navOpenIfSmallScreen, setNavOpenIfSmallScreen] = React.useState(false);
   const location = useLocation();
+  const isSmallScreen = useMatchMedia('(max-width: 1440px)');
 
   const open = React.useCallback(() => {
-    setNavOpen(true);
-  }, []);
+    setNavOpenIfSmallScreen(true);
+    if (!isSmallScreen) {
+      setNavOpenIfLargeScreen(true);
```

Comment:
Does this mean that if I open the left nav with a small viewport and then I make it larger that it gets hidden?

Reply:
If you haven't previously set it to be permanently open in the wide viewport, then yes. (This has been the case since we first introduced the differing behavior based on viewport size.)